symengine/basic_to_upoly.cpp

```diff
@@ -0,0 +1,363 @@
+#include <symengine/visitor.h>
+
+namespace SymEngine
+{
+
+// returns the integer if `x` is a positive integer else
+// returns `-1` if negative integer, `0` if not an integer
+inline int positive_integer(const RCP<const Basic> &x)
+{
+    if (is_a<const Integer>(*x)) {
+        int tmp = rcp_static_cast<const Integer>(x)->as_int();
+        if (tmp > 0)
+            return tmp;
+        else
+            return -1;
```

Comment:
`return rcp_static_cast<const Integer>(x)->as_int();` does the same thing right?